engine/simulationcraft.hpp

```diff
@@ -6360,12 +6363,19 @@ struct real_ppm_t
     if ( last_trigger_attempt == player -> sim -> current_time() )
       return false;
 
-    bool success = player -> rng().roll( proc_chance( player, rppm, last_trigger_attempt, last_successful_trigger, scales_with ) );
-
-    last_trigger_attempt = player -> sim -> current_time();
+    double chance = proc_chance( player, rppm, last_trigger_attempt,
+                                 last_successful_trigger, scales_with );
+    bool success = player -> rng().roll( chance );
 
     if ( success )
+    {
+      first_proc_occurred = true;
       last_successful_trigger = player -> sim -> current_time();
+    }
+    // Due to a bug, last attempt to trigger is fixed until first proc
```

Comment:
It will most likely be fixed in the upcoming biggest patch - 7.0. There is no mechanism needed because all we need to do is avoid merging this change into `legion-dev`.